src/native/monodroid/archive-dso-stub-config.hh.in

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <cstdint>
+
+namespace xamarin::android {
+	struct ArchiveDSOStubConfig
+	{
+		static inline constexpr size_t PayloadSectionAlignment = @ARCHIVE_DSO_STUB_PAYLOAD_SECTION_ALIGNMENT@;
```

Comment:
Why a `.hh.in` file instead of `#ifdef`s or `if constexpr`?

Reply:
Fields are assigned values that are detected by [`CMakeLists.txt`](https://github.com/dotnet/android/pull/9154/files#diff-39a51904cb629ce3d481fc5c48c914e21cd5e608e55f8119f3a1b17f1696fc38R3-R50) after the stub is built.  The stub size, layout etc may differ even within the same target platform if the compiler (e.g. a flag change which affects sections that are present/absent) or linker (e.g. order of section and, thus, size of the entire stub) change, and we must absolutely have the correct values here.  Getting these values require reading the stub ELF file and doing some calculations **after** the payload section is added, since the stub itself is compiled without it (we want to ship the stub without the payload section) - this is what code in `CMakeLists.txt` does.